### PR TITLE
Update Makefile build-cli target

### PR DESCRIPTION
**Pull request by anthonywendt**
Updated to have a build-cli target that will automatically build the appropriate binary based on your system and arch.

### CODE REVIEW

**Comment by anthonywendt** [comment]
This may not be necessary. Just started playing in this repo. Feel free to ignore.

**Review by Noxsios — COMMENTED** [review]
I saw the above comment before I made this review, but still submitting as if this was a more serious PR (but not `requesting changes`). As this PR stands, don't think the benefits outweigh the costs of this change. I leave it up to @UncleGedd .

File: .github/workflows/release.yaml [hunk 1]
```diff
@@ -25,7 +25,7 @@ jobs:
 
       - name: Build CLI
         run: |
-          make build-cli-linux-amd
+          make build-cli
```
[the remark this answers is not in the document]
Comment (UncleGedd) [reply .github/workflows/release.yaml:28]: +1 haven't read through this PR yet but UDS CLI in its current form should run on Linux or Mac arm/amd architectures

**Comment by anthonywendt** [comment]
Do you want me to put any more effort into this? I can close it if not.

**Comment by UncleGedd** [comment]
> Do you want me to put any more effort into this? If so, I can update based on all these comments. I can also close it if not so its not just sitting here. @UncleGedd

Really appreciate you showing me how this works after you guys mentioned it last week. Personally I'm not a fan of all the bash logic, while it results in a simpler `make` cmd, overall I think it's adding more complexity than necessary. Feel free to close for now and we can re-attack sometime in the future if it makes sense

**Pull request closed without merging**